File: rgctl-tests/ecommerce-python/app/models/product.py
from sqlalchemy import ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base
from app.models.base import TimestampMixin


class Product(Base, TimestampMixin):
    __tablename__ = "products"

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    category_id: Mapped[str] = mapped_column(String(36), ForeignKey("categories.id"), index=True)
    name: Mapped[str] = mapped_column(String(255))
    slug: Mapped[str] = mapped_column(String(255), unique=True, index=True)
    description: Mapped[str] = mapped_column(Text, default="")
    price_cents: Mapped[int] = mapped_column(Integer)
    stock: Mapped[int] = mapped_column(Integer, default=0)

    category = relationship("Category", back_populates="products")
    cart_items = relationship("CartItem", back_populates="product", cascade="all, delete-orphan")
    order_items = relationship("OrderItem", back_populates="product")
    reviews = relationship("Review", back_populates="product", cascade="all, delete-orphan")

File: rgctl-tests/ecommerce-go/internal/models/product.go
package models

type Product struct {
	ID          string `gorm:"primaryKey" json:"id"`
	CategoryID  string `gorm:"not null;index" json:"category_id"`
	Name        string `gorm:"not null" json:"name"`
	Slug        string `gorm:"uniqueIndex;not null" json:"slug"`
	Description string `gorm:"not null;default:''" json:"description"`
	PriceCents  int64  `gorm:"not null" json:"price_cents"`
	Stock       int64  `gorm:"not null;default:0" json:"stock"`
	CreatedAt   string `gorm:"not null" json:"created_at"`
}

func (p *Product) SetCreatedAt(v string) { p.CreatedAt = v }
